routes/chars.js

```diff
@@ -4,17 +4,130 @@ var express = require('express'),
     importChar = npmarmory.importChar,
     Char = npmarmory.Char;
 
-// let's see if we can break node here.  ._. prelim logic
-// to import my monk
+// defaults
+var jadeRegionDefault = "US";
+var jadeNameDefault = "Caligraphy";
+var jadeRealmDefault = "Windrunner";
+
+// pre-declare empty so we know where values came from
+var jadeRegion, jadeName, jadeRealm;
+
+var charImporting = function(req, res) {
+
+  //catch tmps from param or POST
+  var tmpRealm = req.tmpRealm, 
+    tmpRegion = req.tmpRegion,
+    tmpName = req.tmpName;
+
+  // if param exist overwrite defaults
+  // I think only (tmp != "undefined") works too
+  if(tmpRegion && tmpRegion != "" && tmpRegion != "undefined"){
+    jadeRegion = tmpRegion;
+  }else{
+    jadeRegion = jadeRegionDefault;
+  };
+  if(tmpName && tmpName != "" && tmpRegion != "undefined"){
+    jadeName = tmpName;
+  }else{
+    jadeName = jadeNameDefault;
+  };
+  if(tmpRealm && tmpRealm != "" && tmpRegion != "undefined"){
+    jadeRealm = tmpRealm;
+  }else{
+    jadeRealm = jadeRealmDefault;
+  };
+
+  //call to npmarmory
+  importChar(jadeRegion,jadeName,jadeRealm, function(data) {
+    //get real char data
+    var theChar = new Char(data);
+
+    //add 'test' params for output onto template
+    theChar.jadeRegion = jadeRegion;
+    theChar.jadeName = jadeName;
+    theChar.jadeRealm = jadeRealm;
+
+    // my tester output text zone
+    theChar.tester = "";
+    theChar.tester += "charRegion: "+req.tmpRegion+" \n";
+    theChar.tester += "charName: "+req.tmpName+" \n";
+    theChar.tester += "charRealm: "+req.tmpRealm+" \n";
+
+
+    // If theChar doesn't have a property (eg. Offhand weapon) then 500 error
+    // also, any not-found error will cause this, so every visible prop needs defaults
+    // unless ofc there's some node thing I dont know about
+    if("Offhand" in theChar){
+      theChar.tester += "Offhand: "+String(theChar.Offhand);
+      if(String(theChar.Offhand) == "No Item"){
+        // theChar.tester += " No Item Detected ";
+        theChar.Offhand = new Object("No Item");
+        theChar.Offhand.weaponInfo = new Object("No Item");
+        theChar.Offhand.weaponInfo.dps = 0.0;
+      }
+    }
```

Comment:
actually, the API should return 0 DMG value for offhand. :) But, I'll look into it. npmarmory should be assigning a default if certain items are missing, but I may have missed that.